proto/vts.proto

```diff
@@ -38,7 +38,8 @@ enum EndorsementType {
 message Endorsement {
     AttestationFormat scheme = 1;
     EndorsementType type = 2;
-    google.protobuf.Struct attributes = 3;
+    uint32   subType = 3;
```

Comment:
The naming convention used in this file uses underscores, so this should be `sub_type`.
Also, why uint?  This better be a string.
Besides, this should be `optional`?
Furthermore (English has so many synonyms!) we should add a comment here to document the "oddities" about this field, at least:
1) the fact that it is "public" but effectively opaque to Veraison
2) the fact that it assumes there is only ever going to be one single level of sub-type nesting
so that future-selves (as well as other Veraison users).

Reply:
Additionally, field numbers must not be arbitrarily changed: see https://developers.google.com/protocol-buffers/docs/proto#assigning

Reply:
Few things:
Makes sense to correct the name to align with proto buffer naming conventions. Will do it!
Why uint: I thought we agreed that the subType is a numeric field, so that the use of numbers does not need to expose the exact sematic of what is on the interface?

Besides why optional: I think, keeping it mandatory does not hurt? It makes the interface consistent. Those plugins who do not want to use it can leave it, will have the default value on the interface ?

I can add comments, I have no issues with that!

Field Number not to be changed: That comment seems more for a finished product released in the field.

Ours is still in the development and debugging face and at the point where changing the sequence is not going to hurt us in any way! Otherwise adding it in the end makes it more difficult to comprehend!

Reply:
> Why uint: I thought we agreed that the subType is a numeric field, so that the use of numbers does not need to expose the exact sematic of what is on the interface?

not sure how that would make a difference, with the disadvantage of making it less readable.

> Besides why optional: I think, keeping it mandatory does not hurt? It makes the interface consistent. Those plugins who do not want to use it can leave it, will have the default value on the interface ?

but that is the exact definition of optional, isn't it?

> Field Number not to be changed: That comment seems more for a finished product released in the field.
> 
> Ours is still in the development and debugging face and at the point where changing the sequence is not going to hurt us in any way! Otherwise adding it in the end makes it more difficult to comprehend!

it's not critical now but it's not good practice either.  Let's try and "do the right thing."

Reply:
> 
I am fine with using string for the semantics if we all agree.

For field numbering:

For numbering the field here the right thing is to define it next to type and number it sequentially as we are in development phase, rather than in the end which is difficult for readers to comprehend and connect with the main type as it hinders readability as well..

So I am all for changing the numbers, as I said earlier, it is still in the development phase and it is perfectly fine to modify the interface at this stage

Reply:
> For field numbering:
> 
> For numbering the field here the right thing is to define it next to type and number it sequentially as we are in development phase, rather than in the end which is difficult for readers to comprehend and connect with the main type as it hinders readability as well..

how so?  numbers are for the wire, not for the reader.  The reader reads the field names.

Reply:
> > For field numbering:
> > For numbering the field here the right thing is to define it next to type and number it sequentially as we are in development phase, rather than in the end which is difficult for readers to comprehend and connect with the main type as it hinders readability as well..
> 
> how so? numbers are for the wire, not for the reader. The reader reads the field names.

I meant the interface in .proto should be logical and readers should be able to comprehend it properly.
Keeping type and sub_type together enables that readability.

Also regarding optionality : Please see: https://github.com/protocolbuffers/protobuf/issues/2497

Reply:
> I meant the interface in .proto should be logical and readers should be able to comprehend it properly. Keeping type and sub_type together enables that readability.

Position is orthogonal to numbering.
 
> Also regarding optionality : Please see: [protocolbuffers/protobuf#2497](https://github.com/protocolbuffers/protobuf/issues/2497)

For proto3 the features are here: https://developers.google.com/protocol-buffers/docs/proto3#specifying_field_rules

Reply:
![image](https://user-images.githubusercontent.com/26571863/203309652-adb979f5-44a3-4a30-bf9d-e5ce4f325f58.png)

Reply:
Seems we cannot use optional as currently protoc-gen-go-json does not support optional fields

https://github.com/mitchellh/protoc-gen-go-json/issues/13

Reply:
Also verified, that if the field is not explicitly set, then the interface treats it as optional and omits the same on the other end of the interface!

Reply:
There is a difference between `singular` and `optional` (cfr. the proto3 docs linked above)